docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -1283,72 +1404,71 @@ spring:
+==== The Redis `RateLimiter`
 
-The `stripVersionMode` parameter has the following possible values: `NEVER_STRIP`, `AS_IN_REQUEST` (default), and `ALWAYS_STRIP`.
+The Redis implementation is based off of work done at https://stripe.com/blog/rate-limiters[Stripe].
+It requires the use of the `spring-boot-starter-data-redis-reactive` Spring Boot starter.
 
- * `NEVER_STRIP`: The version is not stripped, even if the original request path contains no version.
- * `AS_IN_REQUEST` The version is stripped only if the original request path contains no version.
- * `ALWAYS_STRIP` The version is always stripped, even if the original request path contains version.
+The algorithm used is the https://en.wikipedia.org/wiki/Token_bucket[Token Bucket Algorithm].
 
-The `hostValue` parameter, if provided, is used to replace the `host:port` portion of the response `Location` header.
-If it is not provided, the value of the `Host` request header is used.
+The `redis-rate-limiter.replenishRate` property is how many requests per second you want a user to be allowed to do, without any dropped requests.
+This is the rate at which the token bucket is filled.
 
-The `protocolsRegex` parameter must be a valid regex `String`, against which the protocol name is matched.
-If it is not matched, the filter does nothing.
-The default is `http|https|ftp|ftps`.
+The `redis-rate-limiter.burstCapacity` property is the maximum number of requests a user is allowed to do in a single second.
+This is the number of tokens the token bucket can hold.
+Setting this value to zero blocks all requests.
 
-=== The `RewriteResponseHeader` `GatewayFilter` Factory
+The `redis-rate-limiter.requestedTokens` property is how many tokens a request costs.
+This is the number of tokens taken from the bucket for each request and defaults to `1`.
 
-The `RewriteResponseHeader` `GatewayFilter` factory takes `name`, `regexp`, and `replacement` parameters.
-It uses Java regular expressions for a flexible way to rewrite the response header value.
-The following example configures a `RewriteResponseHeader` `GatewayFilter`:
+A steady rate is accomplished by setting the same value in `replenishRate` and `burstCapacity`.
+Temporary bursts can be allowed by setting `burstCapacity` higher than `replenishRate`.
+In this case, the rate limiter needs to be allowed some time between bursts (according to `replenishRate`), as two consecutive bursts will result in dropped requests (`HTTP 429 - Too Many Requests`).
+The following listing configures a `redis-rate-limiter`:
+
+Rate limits bellow `1 request/s` are accomplished by setting `replenishRate` to the wanted number of requests, `requestedTokens` to the timespan in seconds and `burstCapacity` to the product of `replenishRate` and `requestedTokens`, e.g. setting `replenishRate=1`, `requestedTokens=60` and `burstCapacity=60` will result in a limit of `1 request/min`.
```

Comment:
Change to:

Rate limits below `1 request/s` are accomplished by setting `replenishRate` to the wanted number of requests, `requestedTokens` to the timespan in seconds, and `burstCapacity` to the product of `replenishRate` and `requestedTokens`. For example, setting `replenishRate=1`, `requestedTokens=60`, and `burstCapacity=60` results in a limit of `1 request/min`.

(I would put each sentence on its own line, too.)